src/views/List.jsx

```diff
@@ -1,14 +1,42 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchString, setSearchString] = useState('');
+
+	const handleChange = (e) => {
+		setSearchString(e.target.value);
+	};
+
+	const handleClick = (e) => {
+		e.preventDefault();
+		setSearchString('');
+	};
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+			<form>
+				<label htmlFor="searchString">
+					Search:
+					<input
+						type="text"
+						id="searchString"
+						name="searchString"
+						value={searchString}
+						onChange={handleChange}
+					/>
+				</label>
+				{searchString ? <button onClick={handleClick}>x</button> : ''}
+			</form>
+
 			<ul>
 				{data && data.length > 0 ? (
-					data.map((item, id) => <ListItem key={id} name={item.name} />)
+					data
+						.filter((d) => d.name.includes(searchString))
```

Comment:
Search is currently case sensitive, suggestion to set search string and items .toLowerCase/.toUpperCase